src/ProvidedTypes.fs

```diff
@@ -6652,12 +6677,36 @@ namespace ProviderImplementation.ProvidedTypes.AssemblyReader
 
         let GetReaderCache () = ReadOnlyDictionary(readerCache)
 
+        // Auto-clear the cache every 30.0 seconds.
+        // We would use System.Runtime.Caching but some version constraints make this difficult.
+        let enableAutoClear = try Environment.GetEnvironmentVariable("FSHARP_TPREADER_AUTOCLEAR_OFF") = null with _ -> true
+        let clearSpanDefault = 30000

```

Comment:
What's the significance of 30 seconds compared with other values? When we consider that FCS re-checks frequently, how often would this re-allocate the large amount of data (400+ MB) on the LOH in normal usage?